c2rust-analyze/src/rewrite/statics.rs

```diff
@@ -0,0 +1,55 @@
+use crate::context::{PermissionSet, PointerId};
+use crate::rewrite::Rewrite;
+use crate::{GlobalAnalysisCtxt, GlobalAssignment};
+use rustc_hir::{ItemKind, Mutability, Node};
+use rustc_span::Span;
+
+/// For every static, if its write permission does not match its declared mutability, emit a rewrite
+/// changing the declaration to match observed/analyzed usage.
+pub fn gen_static_rewrites<'tcx>(
+    gacx: &GlobalAnalysisCtxt<'tcx>,
+    gasn: &GlobalAssignment,
+) -> Vec<(Span, Rewrite)> {
+    let mut hir_rewrites = Vec::new();
+    for (did, lty) in gacx.static_tys.iter() {
+        // The map of statics and their ty + permissions tracks statics by did; map this to an Item
+        // node to look at the static's spans and declared mutability.
+        let item = if let Some(Node::Item(item)) = gacx.tcx.hir().get_if_local(*did) {
+            item
+        } else {
+            panic!("def id {:?} not found", did);
+        };
+        if let ItemKind::Static(_ty, mut_, _body_id) = item.kind {
```

Comment:
Variables of type `Mutability` are more commonly named `mutbl` (both here and in rustc)